src/Cryptol/TypeCheck/SimpType.hs

```diff
@@ -113,7 +113,13 @@ tSub :: Type -> Type -> Type
 tSub x y
   | Just t <- tOp TCSub (op2 nSub) [x,y] = t
   | tIsInf y  = tError (tf2 TCSub x y)
+
   | tIsInf x = x
+    {- This assumes that `y` is finite and not error.  The first should
+       follow from the typing on `tSub`, which asserts that the second argumet
+       is finite and less than the first;  the second should have been handleed
+       by the first equation above `tOp`. -}
```

Comment:
I'm not sure which equation this refers to. There is a use of `tOp` in the first equation, but there are no equations above it.